### ISSUE TITLE
Many to many and on match destroy in `manage_relationship`

### ISSUE THREAD

**Issue by vonagam** [ash-project/ash#859]
I think `on_match: :destroy` in `manage_relationship` is incorrect for `many_to_many`.

Right now it destroys a join resource. Instead it should destroy both a join and a destination like other types of relationships do.

If one wants to destroy only a join resource then they should use `:unrelate`.

Destroy is implemented correctly for `on_missing` where `:destroy` destroys both and `:unrelate` only join one. (Though there is a mistake in documentation that states that it is possible to pass `[:join, :keys]` with a long form.) That's the only other case with support for `:destroy` so it is strange that they are inconsistent.

**Comment by zachdaniel** [comment]
Yeah, I think you're correct. It does potentially make sense to be able to pass join keys to a destroy action on a join resource though, since destroy actions can technically take arguments.

**Comment by vonagam** [comment]
But not for `on_missing` since there is no input for those.

**Comment by jaredsrice** [comment]
Hi, I’m new to contributing to Ash and noticed this is labeled good first issue, but I also see #2638 is already open. Is there any smaller follow-up, test case, or documentation update related to this that would be useful for a first contribution?

**Comment by zachdaniel** [comment]
👋 thank you for reminding me of that PR. I don't believe there is any needed followup but I very much appreciate you bringing it back to my attention!
